### DISCUSSION TITLE
How to fetch a list of shipping countries and country areas (Saleor 3.0)

### DISCUSSION THREAD

**Discussion by Juleszz** [saleor/saleor#7922] [Q&A]
Hey guys,

what is the best practice for fetching possible shipping **countries** and country **areas**?
During checkout the customer has to select the shipping country and ideally this should be the same list as shipping zones - or let's say it should be a country where a shipping method exists. However I have no permissions for fetching shipping zones and the documentation points out that these will never be exposed to the customer directly. 

Thanks for your help!

**Comment by karolkielecki** [comment]
@Juleszz Hi! I'm not sure if I understood you correctly. What exactly would you like to achieve? Would you like to let customers to see all available shipping zones? If you provide some use-case, that would be great.

**Reply by Juleszz** [reply]
During checkout the customer normally selects a shipping country. Basically what I'd like to have as that this list of countries is populated by a query e.g. if our company even ships to this country. Further it would be helpful to also query possible country areas within this country e.g. states for the US. 

I could create such a list manually but I thought there would be an easier way of just fetching all this data so it will also be updated once we create a new shipping zone and expand into more countries.

**Reply by karolkielecki** [reply]
You can limit the shipping zone to shipping addresses within or outside of certain postal code ranges--do you think that would work for you?

**Reply by Juleszz** [reply]
The problem is fetching those zones or all shipping countries + states from the frontend.
